### PR TITLE
Revert "remove extraneous transform tests"

### PR DESCRIPTION
**Pull request by jsetoain**
This reverts commit 6dab630c74084f49d13322f7dd81829cf68466df.

There's nothing "extraneous" about these tests.

### CODE REVIEW

**Comment by makslevental** [comment]
@jsetoain sorry maybe extraneous isn't the right word but I saw that they don't test `aie` functionality (only upstream). In addition, as you can see, they now error.

**Comment by jsetoain** [comment]
> @jsetoain sorry maybe extraneous isn't the right word but I saw that they don't test `aie` functionality (only upstream). In addition, as you can see, they now error. I guess it's fine if they stay but I wasn't sure the benefit against the (current) cost.

They test the `transform.structured.vectorize_contraction` transformation, which was added together with the tests themselves (f388ebe06f5d9d7262eae190c82df2a814ebb2bc). It's not upstream functionality, it's ours.

File: lib/Dialect/AIEVec/TransformOps/AIEVecTransformOps.cpp [hunk 1]
```diff
@@ -239,7 +231,7 @@ DiagnosedSilenceableFailure transform::VectorizeContractionOp::applyToOne(
   for (unsigned i = 0; i < numOuterMostDims; i++)
     remOuterDims.push_back(getAffineDimExpr(i, ctx));
   unsigned numResults = indexingMaps[0].getNumResults();
-  llvm::ArrayRef<int64_t> positions = {numResults - 2, numResults - 1};
+  SmallVector<int64_t> positions = {numResults - 2, numResults - 1};
```
Comment (makslevental) [inline lib/Dialect/AIEVec/TransformOps/AIEVecTransformOps.cpp:234]: Sorry this bit you :(

**Review by makslevental — APPROVED** [review]

File: lib/Dialect/AIEVec/TransformOps/AIEVecTransformOps.cpp (hunk 1 above)
Comment (jsetoain) [reply lib/Dialect/AIEVec/TransformOps/AIEVecTransformOps.cpp:234]: It's all good, you solved it yourself so... 😄

**Pull request merged**